### DISCUSSION TITLE
Multi Preferences for Vaccines

### DISCUSSION THREAD

**Discussion by shailesh** [pallupz/covid-vaccine-booking#163] [Ideas]
Currently, we have a single or all preference. Now that we have more than two vaccines, can we make it to multi preferences?

```
def get_vaccine_preference():
    print(
        "It seems you're trying to find a slot for your first dose. Do you have a vaccine preference?"
    )
    preference = input(
        "Enter 0 for No Preference, 1 for COVISHIELD, 2 for COVAXIN, or 3 for SPUTNIK V. Default 0 : "
    )
    preference = int(preference) if preference and int(preference) in [0, 1, 2, 3] else 0

    if preference == 1:
        return "COVISHIELD"
    elif preference == 2:
        return "COVAXIN"
    elif preference == 3:
        return "SPUTNIK V"
    else:
        return None
```

**Comment by vasugamdha** [comment]
This is a redundant discussion thread. [Please continue here](https://github.com/pallupz/covid-vaccine-booking/discussions/160#discussion-3368017)

**Comment by pallupz** [comment]
@shailesh its possible but right now we are filtering using query parameters.. instead we'll have to fetch all results and filter it out

**Reply by pallupz** [reply]
its a bit more work.. i will try to get to it over the weekend
